test-cypress/integration/setup/user-and-access.spec.js

```diff
@@ -59,4 +47,241 @@ describe('User and Access', () => {
         // Date created should be visible
         cy.get('@user').find('.date-created').should('be.visible');
     });
+
+    it('Create user - read-only and verify that security works', () => {
+        cy.wait(500); //sometimes repositories list cannot be fetched.
+        getCreateNewUserButton().click();
+        getUsernameField().type(user);
+        getPasswordField().type(password);
+        getConfirmPasswordField().type(password);
+        getUserRoleRadioButton("roleUser").click();
+        getRepositoryRights(repositoryId, "read").click();
+        getCreateButton()
+            .should('be.visible')
+            .click();
+        findUserInTable(user);
```

Comment:
I guess we can extract these commands in  function like `createUser(role)` and reuse it in the other tests below.

Reply:
Fixed with commit d5a40ea358aa33047d813e1f1b223285da550018